src/Symfony/Component/TypeInfo/TypeResolver/StringTypeResolver.php

```diff
@@ -223,7 +223,17 @@ private function getTypeFromNode(TypeNode $node, ?TypeContext $typeContext): Typ
         }
 
         if ($node instanceof UnionTypeNode) {
-            return Type::union(...array_map(fn (TypeNode $t): Type => $this->getTypeFromNode($t, $typeContext), $node->types));
+            $types = [];
+
+            foreach ($node->types as $nodeType) {
+                $types[] = $type = $this->getTypeFromNode($nodeType, $typeContext);
+
+                if ($type instanceof BuiltinType && TypeIdentifier::MIXED === $type->getTypeIdentifier()) {
+                    return Type::mixed();
+                }
```

Comment:
You can early return 
```suggestion
                if ($type instanceof BuiltinType && TypeIdentifier::MIXED === $type->getTypeIdentifier()) {
                    return Type::mixed();
                }
                
                $types[] = $type = $this->getTypeFromNode($nodeType, $typeContext);
```

Reply:
The `$type` must be computed (thanks to `getTypeFromNode`) before the `if`. Otherwise, it won't return a mixed if the last type is the mixed one.

Reply:
Ah, didn't see this subtil `= $type =`